File: 2020/02/main.cpp
#include <iostream>
#include <sstream>
#include <string>
#include <fstream>
#include <vector>

using namespace std;

int p1(const vector<string>& lines) {
    int i = 0;
    for(auto line : lines) {
        istringstream iss(line);
        int l, h;
        string sub;
        iss >> sub;
        l = stoi(sub.substr(0, sub.find('-')));
        h = stoi(sub.substr(sub.find('-')+1));
        iss >> sub;
        char c = sub[0];
        string p;
        iss >> p; 
        int n = count(p.cbegin(), p.cend(), c);
        if (n >=l && n <= h)
            i++;
    }

    return i;
}

int main(int argc, char** argv) {

    fstream f("input.txt", ios::in);
    if (!f.is_open()) return -1;

    vector<string> lines;
    string line;
    while(getline(f, line)) {
        lines.push_back(line);
    }

    cout << p1(lines) << endl;

}

File: 2020/02/main.py
def p1(lines):
    v = []
    for line in lines:
        l, h = line.split()[0].split('-')
        l = int(l)
        h = int(h)
        c = line.split()[1][0]
        p = line.split()[2]
        n = p.count(c)
        if n >= l and n <= h:
            v.append(p)

    print(len(v))

def p2(lines):
    v= []
    for line in lines:
        l, h = line.split()[0].split('-')
        l = int(l)
        h = int(h)
        c = line.split()[1][0]
        p = line.split()[2]
        if (p[l-1] == c) != (p[h-1] == c):
            v.append(p)
    print(len(v))

if __name__ == '__main__':
    with open('input.txt', 'r') as f:
        lines = f.read().splitlines()

    p1(lines)
    p2(lines)

